src/main/java/com/apajac/acolhimento/services/PersistirDoadorServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.apajac.acolhimento.services;
+
+import com.apajac.acolhimento.domain.dtos.DoadorDTO;
+import com.apajac.acolhimento.domain.entities.DoadorEntity;
+import com.apajac.acolhimento.domain.enums.AuditoriaEnum;
+import com.apajac.acolhimento.exceptions.BusinessException;
+import com.apajac.acolhimento.exceptions.NotFoundException;
+import com.apajac.acolhimento.repositories.DoadorRepository;
+import com.apajac.acolhimento.services.interfaces.AuditoriaService;
+import com.apajac.acolhimento.services.interfaces.PersistirDoadorService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Objects.isNull;
+
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class PersistirDoadorServiceImpl implements PersistirDoadorService {
+
+    private final DoadorRepository doadorRepository;
+
+    private final AuditoriaService auditoria;
+
+    @Override
+    public void persistirDoador(DoadorDTO doadorDTO) {
+        if(isNull(doadorDTO.getId())){
+            try {
```

Comment:
O If deveria estar dentro do TryCatch pois se a clausula do If não for atendida o TryCatch não será chamado.